### ISSUE TITLE
[BUG/ISSUE] Lumping species is very slow in benchmark code

### ISSUE THREAD

**Issue by yantosca** [geoschem/gcpy#166]
The "Adding lumped species to dataset" in the benchmark code is very slow.  We should investigate if there are ways to speed it up.  It may be just due to how we are merging the new DataArrays into the DataSet objects.

**Comment by yantosca** [comment]
@lizziel says that the slowdown is most pronounced when computing the AnnualMean in 1-yr benchmarks.

Also tagging @laestrada

**Comment by yantosca** [comment]
This is now fixed with commit: df1dae2b716f3fdecb5da3c4da543006a44f8582.  

The speedup with the new algorithm is:
```console
add_lumped_species_to_dataset
before:  1.4314453601837158 sec
after:   0.3670790195465088 sec
```

**Comment by yantosca** [comment]
Also I forgot to mention that identical results were obtained w/r/t the prior lumping algorithm.

**Comment by yantosca** [comment]
We can close this issue since PR #164 is now merged into the dev branch.
